app/web/public/demo-worker.js

```diff
@@ -0,0 +1,625 @@
+  let vLen = csg.sides.length * 6
+
+  var vertices = new Float32Array(vLen)
+  csg.sides.forEach((side,idx)=>{
+    let i = idx * 6
+    setPoints(vertices, side[0], i)
+    setPoints(vertices, side[1], i+3)
+  })
+  return {vertices, type:'lines'}
+
+}
+
+function CSGCached(func, data, cacheKey, transferable){
+  cacheKey = cacheKey || data
+
+  let geo = CSGToBuffers.cache.get(cacheKey)
+  if(geo) return geo
+
+  geo = func(data)
+
+  // fill transferable array for postMessage optimization
+  if(transferable){
+    const {vertices, indices} = geo
+    transferable.push(vertices)
+    if(indices) transferable.push(indices)
+  }
+
+    CSGToBuffers.cache.set(cacheKey, geo)
+    return geo
+}
+
+function CSGToBuffers(csg, transferable){
+  let obj
+
+  if(csg.polygons) obj = CSGCached(CSG2Vertices,csg,csg.polygons, transferable)
+  if(csg.sides && !csg.points) obj = CSGCached(CSG2LineSegmentsVertices,csg,csg.sides, transferable)
+  if(csg.points) obj = CSGCached(CSG2LineVertices,csg,csg.points, transferable)
+
+  return obj
+}
+CSGToBuffers.clearCache = ()=>{CSGToBuffers.cache = new WeakMap()}
+CSGToBuffers.clearCache()
+
+
+
+
+let workerBaseURI
+
+function require(url){
+    url = require.alias[url] || url
+    if(url[0] != '/' && url.substr(0,2) != './' && url.substr(0,4) != 'http') url = 'https://unpkg.com/'+url
+    let exports=require.cache[url]; //get from cache
+    if (!exports) { //not cached
+      let module = requireModule(url)
+      require.cache[url]  = exports = module.exports; //cache obj exported by module
+    }
+    return exports; //require returns object exported by module
+}
+
+function requireFile(url){
+  try{
+    let X=new XMLHttpRequest();
+    X.open("GET", new URL(url,workerBaseURI), 0); // sync
+    X.send();
+    if (X.status && X.status !== 200)  throw new Error(X.statusText);
+    return X.responseText;
+  }catch(e){
+    console.log('problem loading url ',url,'base',workerBaseURI,' error:',e.message)
+    throw e
+  }
+}
+
+function requireModule(url, source){
+    try {
+        const exports={};
+        if(!source) source = requireFile(url)
+        const module = { id: url, uri: url, exports:exports, source }; //according to node.js modules
+        // fix, add comment to show source on Chrome Dev Tools
+        source="//@ sourceURL="+url+"\n" + source;
+        //------
+        const anonFn = new Function("require", "exports", "module", source); //create a Fn with module code, and 3 params: require, exports & module
+        anonFn(require, exports, module); // call the Fn, Execute the module
+        return module
+    } catch (err) {
+        console.error("Error loading module "+url, err.message);
+        throw err;
+    }
+}
+
+require.cache = {}
+require.alias = {}
+
+
+const initCanvas = (canvas, callback)=>{
+
+  // convert HTML events (mouse movement) to viewer changes
+  let lastX = 0
+  let lastY = 0
+
+  let pointerDown = false
+
+  const moveHandler = (ev) => {
```

Comment:
I don't quiet follow how these handlers fit in? especially since this runs in the worker. Could you explain it a little?

Reply:
it is code when canvas is used directly 
The worker has 3 parts:
 - renderer
 - orbit controls
 - script eval

I was not sure if we would go in direction of direct canvass access. for now, this integration only "script eval" part is needed.

I curently copy/pasta the worker from my other prototype. After I refine the prototype worker, I will have a version that only has "script eval" part.

Reply:
okay yup, cool.